flag_engine/identities/models.py

```diff
@@ -51,6 +63,35 @@ def in_segment(self, segment: SegmentModel) -> bool:
             for rule in segment.rules
         )
 
+    def find_trait_index(self, trait: TraitModel) -> typing.Optional[int]:
+        try:
+            return self.identity_traits.index(trait)
+        except ValueError:
+            return None
+
+    def update_traits(self, traits: typing.List[TraitModel]):
+        for trait in traits:
+            trait_index = self.find_trait_index(trait)
+            # Delete the trait if the value is None
+            if trait.trait_value is None and trait_index is not None:
+                    self.identity_traits.pop(trait_index)
+                    continue
+            # Update the trait if it already exists
+            if trait_index is not None:
+                self.identity_traits[trait_index] = trait
+            else:
+                self.identity_traits.append(trait)
+        return self.identity_traits
+
+    def generate_traits(self, traits: typing.List[TraitModel], persist: bool = False):
```

Comment:
I don't really know what this method is doing? It seems that it could be simplified to: 

```python
if persist:
    self.identity_traits = trait_models
return trait_models
```

If that's the case, do we need this method? Can't we just handle that logic elsewhere?